### PR TITLE
slither-analyzer 0.8.3 (new formula)

### PR DESCRIPTION
**Pull request by elopez**
Slither is a Solidity static analysis framework written in Python 3. It runs a suite of vulnerability detectors, prints visual information about contract details, and provides an API to easily write custom analyses.

Full disclosure: I work at @trailofbits and contribute to Slither and other open-source company tools.

-----

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Review by chenrui333 — DISMISSED** [review]

File: Formula/slither-analyzer.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  license "AGPL-3.0-only"
+  head "https://github.com/crytic/slither.git", branch: "master"
+
+  depends_on "python@3.10"
+  depends_on "solc-select"
+
+  resource "crytic-compile" do
+    url "https://github.com/crytic/crytic-compile/archive/refs/tags/0.2.3.tar.gz"
+    sha256 "b41cc3ca6e43b70c2257dba5c1e794ca0881577f908acee9bc5bdfb132792e30"
+  end
```
Comment (carlocab) [inline Formula/slither-analyzer.rb:17]: We already have a `crytic-compile` formula -- can't we use that?

Comment (elopez) [reply Formula/slither-analyzer.rb:17]: Hi @carlocab! crytic-compile is both an standalone program that you can run, as well as a library that you can import and use. In this case, Slither imports and calls things from crytic-compile ([example](https://github.com/crytic/slither/blob/ce9dbf650d7acfee51ddafd844929f4e8d345672/slither/core/slither_core.py#L13-L14))

My understanding is that each of these formulas creates a separate python venv, so importing across them wouldn't work. I may be missing something though, I'm pretty new to brew packaging. How do you normally handle these?

Comment (carlocab) [reply Formula/slither-analyzer.rb:17]: You can write a `.pth` file into this formula's private `site-packages` directory that will allow it to access `crytic-compile`'s Python libraries.

See `ansible-lint` for an example (though the syntax is a little outdated):

https://github.com/Homebrew/homebrew-core/blob/3836ab56271f54f57f08923e00d8a18ffc112c4a/Formula/ansible-lint.rb#L175-L181

Comment (elopez) [reply Formula/slither-analyzer.rb]: Awesome, thanks! I've now added crytic-compile as a dependency, and write the pth during this formula's installation to add the crytic-compile library

File: Formula/slither-analyzer.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  resource "wcwidth" do
+    url "https://files.pythonhosted.org/packages/89/38/459b727c381504f361832b9e5ace19966de1a235d73cdbdea91c771a1155/wcwidth-0.2.5.tar.gz"
+    sha256 "c4d647b99872929fdb7bdcaa4fbe7f01413ed3d98077df798530e5b04f116c83"
+  end
+
+  def install
+    virtualenv_install_with_resources
+    site_packages = Language::Python.site_packages(Formula["python@3.10"].opt_bin/"python3")
```
Comment (carlocab) [inline Formula/slither-analyzer.rb:32]:
```suggestion
    site_packages = Language::Python.site_packages("python3.10")
```

File: Formula/slither-analyzer.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  resource "prettytable" do
+    url "https://files.pythonhosted.org/packages/10/88/ef38a6e4bc375600d3031e405a8d3b3dc4a154fccffd21d5d06e66c96230/prettytable-3.3.0.tar.gz"
+    sha256 "118eb54fd2794049b810893653b20952349df6d3bc1764e7facd8a18064fa9b0"
+  end
+
+  resource "pysha3" do
+    url "https://files.pythonhosted.org/packages/73/bf/978d424ac6c9076d73b8fdc8ab8ad46f98af0c34669d736b1d83c758afee/pysha3-1.0.2.tar.gz"
+    sha256 "fe988e73f2ce6d947220624f04d467faf05f1bbdbc64b0a201296bb3af92739e"
+  end
```
Comment (carlocab) [inline Formula/slither-analyzer.rb:23]: I think this might not be needed anymore, since it's in `crytic-compile`'s virtualenv.

Comment (elopez) [reply Formula/slither-analyzer.rb]: Makes sense, yep!

File: Formula/slither-analyzer.rb (hunk 2 above)
Comment (elopez) [reply Formula/slither-analyzer.rb]: Applied the suggested change 👍

**Review by carlocab — APPROVED** [review]

File: Formula/slither-analyzer.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
+class SlitherAnalyzer < Formula
+  include Language::Python::Virtualenv
+
+  desc "Solidity static analysis framework written in Python 3"
+  homepage "https://github.com/crytic/slither"
+  url "https://github.com/crytic/slither/archive/refs/tags/0.8.3.tar.gz"
```
Comment (carlocab) [inline Formula/slither-analyzer.rb:6]: Would be nice for these tarballs to get published on PyPI. It's a lot easier for `brew` to automatically update `resource`s that way.

**Review by carlocab — APPROVED** [review]
Thanks again!

File: Formula/slither-analyzer.rb (hunk 4 above)
Comment (elopez) [reply Formula/slither-analyzer.rb:6]: I think Slither used to have them, but I don't know what happened there. I'll ask to see if we can have them back on the next release.

**Pull request closed without merging**